packages/send_email/utils.py

```
@@ -0,0 +1,81 @@
+import json
+import os
+from datetime import datetime
+
+import boto3
+from botocore.exceptions import ClientError
+from lr_retry import retry_func
+from spine_aws_common.logger import Logger
+
+logger = Logger()
+
+# TODO
+# EMAIL_BUCKET = os.environ["EMAIL_BUCKET"]
+
+
+class InvalidCharacter(Exception):
+    pass
+
+
+def upload(email, filename, global_args):
+    # TODO code for running locally, incld global_args - pls ignore
+    if global_args.running_locally:
+        path = "emails"
+        if not os.path.isdir(path):
+            os.makedirs(path)
+        filename = os.path.join(path, filename)
+
+        with open(filename, "w") as f:
+            f.write(email)
+        return True
+
+    else:
+
+        EMAIL_BUCKET = os.environ["EMAIL_BUCKET"]
+
+        s3 = boto3.client("s3")
+
+        try:
+            retry_func(
```

Is this required? I believe the AWS SDK has some level of retries built in?

Can we just return an error and let the calling function decide to retry or not?